apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/ActionButton/ActionButton.tsx

```diff
@@ -22,46 +26,71 @@ export function ActionButton({
   const { t } = useTranslation('apps-journeys-admin')
   const updateEdge = useUpdateEdge()
 
+  function getIcon(action?: Action | null): ReactNode {
+    switch (action?.__typename) {
+      case 'LinkAction':
+        return (
+          <Tooltip title={action.url} placement="left" arrow>
+            <LinkIcon
+              sx={{ fontSize: 10, position: 'absolute', left: -20, top: 7 }}
+            />
+          </Tooltip>
+        )
+      case 'EmailAction':
+        return (
+          <Tooltip title={action.email} placement="left" arrow>
+            <EmailIcon
+              sx={{ fontSize: 10, position: 'absolute', left: -20, top: 8 }}
+            />
+          </Tooltip>
+        )
+      default:
+        return <></>
+    }
+  }
+
+  function getSourceConnection(block): boolean {
```

Comment:
rename please, naming could be more clear about what it's doing

```suggestion
  function hasConnection(block): boolean {
```